test/read-stream.js

```diff
@@ -54,6 +54,88 @@ describe( 'BlockMap.ReadStream', function() {
 
   })
 
+  it( 'should not close the fd if autoClose = false', function( done ) {
+
+    var filename = path.join( __dirname, '/data/bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var fd = fs.openSync( filename, 'r' )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( null, blockMap, {
+      fd: fd,
+      autoClose: false,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
+        assert.ok( fs.fstatSync( readStream.fd ) )
+        fs.closeSync( readStream.fd )
+        done()
+      })
+
+  })
+
+  it( 'should start reading at `start`, if specified', function( done ) {
+
+    var filename = path.join( __dirname, '/data/padded-bmap.img' )
+    var blockMap = BlockMap.create( require( './data/version-2.0' ) )
+    var blockCount = 0
+    var readStream = new BlockMap.ReadStream( filename, blockMap, {
+      start: 4096,
+    })
+
+    readStream
+      .on( 'data', ( block ) => {
+        blockCount++
+        assert.ok( block.address != null, 'block address missing' )
+        assert.ok( block.position != null, 'block position missing' )
+      })
+      .once( 'error', done )
+      .once( 'end', function() {
+        assert.equal( this.blocksRead, blockMap.mappedBlockCount, 'blocksRead mismatch' )
+        assert.equal( this.bytesRead, blockMap.mappedBlockCount * blockMap.blockSize, 'bytesRead mismatch' )
+        assert.equal( this.rangesRead, blockMap.ranges.length, 'rangesRead mismatch' )
+        assert.equal( blockCount, blockMap.mappedBlockCount, 'actual blocks read mismatch' )
```

Comment:
Start & end are byte offsets, meant for reading a range in a container – this should not affect mapped block reading. Also, you can't start at an arbitrary block anyways, as the bmap has mapped block *ranges*, and the checksum validation would fail.
Of course the number of blocks read can still be limited by `end`, as it will just stop once it hits the end, and only validate fully read ranges. But that's not what one should be doing (even though I did do it in the tests) to achieve this.

If you'd want to skip blocks, you could put a transform in between the readable & writable omitting blocks, i.e. if you'd want to skip every block with an address not divisible by 2:

```js
var transform = new stream.Transform({
  transform( block, _, next ) {
    if( block.address % 2 === 0 ) next( null, block )
    else next()
  }
})

BlockMap.createReadStream( 'mapped.img', blockMap )
  .pipe( transform )
  .pipe( blockWriteStream )
```